analyzers/tests/SonarAnalyzer.UnitTest/TestCases/InfiniteRecursion.CSharp11.cs

```diff
@@ -14,6 +14,66 @@ interface InfiniteRecursion
 
     public class Addition : IAdditionOperators<Addition, Addition, Addition>
     {
-        public static Addition operator +(Addition left, Addition right) => left + right; // FN
+        public static Addition operator +(Addition left, Addition right) => left + right; // Noncompliant
+//                                      ^
+    }
+
+    public class Comparison : IComparisonOperators<Comparison, Comparison, Comparison>
+    {
+        public static Comparison operator ==(Comparison? left, Comparison? right)
+//                                        ^^
+        {
+            return left == right;
+        }
+
+        public static Comparison operator !=(Comparison left, Comparison right) => left != right; // Noncompliant
```

Comment:
we only have `// Noncompliant` test cases for operators. I suggest to also add a compliant one (even if weird)
```suggestion
        public static Comparison operator !=(Comparison left, Comparison right) => left == right; // compliant (because we do not support cross-method analysis)
```